src/sentry/lang/javascript/processor.py

```diff
@@ -1576,45 +1784,54 @@ def _fetch_and_cache_sourceview(self, url, debug_id, source_file_type):
+            return sourceview, FetcherSource.URL
 
     def get_or_fetch_sourcemap_cache(self, url=None, debug_id=None):
         """
         Gets from the cache or fetches the SmCache of the file at 'url' or 'debug_id'.
 
         Returns None when a sourcemap corresponding to the file at 'url' is not found.
         """
-        sourcemap_cache, resolved_with_debug_id = self._get_cached_sourcemap_cache(url, debug_id)
+        sourcemap_cache, resolved_with = self._get_cached_sourcemap_cache(url, debug_id)
         if sourcemap_cache is not None:
-            return sourcemap_cache, resolved_with_debug_id
+            return sourcemap_cache, resolved_with
 
         return self._fetch_and_cache_sourcemap_cache(url, debug_id)
 
     def _get_cached_sourcemap_cache(self, url, debug_id):
         if debug_id is not None:
-            rv = self.debug_id_sourcemap_cache.get(debug_id)
-            if rv is not None:
-                return rv, True
+            sourcemap_cache = self.debug_id_sourcemap_cache.get(debug_id)
+            if sourcemap_cache is not None:
+                return sourcemap_cache, FetcherSource.DEBUG_ID
 
         if url is not None:
             # We get the url of the sourcemap from the url of the minified file.
             sourcemap_url = self.minified_source_url_to_sourcemap_url.get(url)
-            rv = self.sourcemap_url_sourcemap_cache.get(sourcemap_url)
-            if rv is not None:
-                return rv, False
 
-        return None, False
+            sourcemap_cache = self.sourcemap_url_new_sourcemap_cache.get(sourcemap_url)
+            if sourcemap_cache is not None:
+                return sourcemap_cache, FetcherSource.URL_NEW
+
+            sourcemap_cache = self.sourcemap_url_sourcemap_cache.get(sourcemap_url)
+            if sourcemap_cache is not None:
+                return sourcemap_cache, FetcherSource.URL
+
+        return None, FetcherSource.NONE
 
     def _fetch_and_cache_sourcemap_cache(self, url, debug_id):
         if debug_id is not None:
-            sourcemap_view = self._handle_debug_id_sourcemap_lookup(debug_id)
-            if sourcemap_view is not None:
-                return sourcemap_view, True
+            sourcemap_cache = self._handle_debug_id_sourcemap_lookup(debug_id)
+            if sourcemap_cache is not None:
+                return sourcemap_cache, FetcherSource.DEBUG_ID
 
         if url is not None:
-            sourcemap_view = self._handle_url_sourcemap_lookup(url)
-            if sourcemap_view is not None:
-                return sourcemap_view, False
+            sourcemap_cache = self._handle_url_sourcemap_lookup(url, use_url_new=True)
+            if sourcemap_cache is not None:
+                return sourcemap_cache, FetcherSource.URL_NEW
+
+            sourcemap_cache = self._handle_url_sourcemap_lookup(url, use_url_new=False)
+            if sourcemap_cache is not None:
+                return sourcemap_cache, FetcherSource.URL
```

Comment:
This needs clarification.

I thought if we have a debug-id, we will not be falling back to "fetching" (aka looking up inside artifact bundles) by url?

Also it not entirely clear to me what `url_new` is supposed to be just by reviewing the code.

Reply:
In this implementation we fallback because we have to support existing symbolication with the old algorithm. The idea of the `url_new` is to have the fetching on the new `ReleaseArtifactBundle` table.
Basically the three fetching logic are:
* `by_debug_id` -> fetches the minified source and sourcemap with the debug id using the `DebugIdArtifactBundle` table.
* `by_url_new` -> fetches all the artifact bundles bound to a `release-dist` pair using the `ReleaseArtifactBundle` table.
* `by_url` -> fetches all the artifact bundles using the `ReleaseFile` table or the internet.